### PR TITLE
Workaround for SDL race condition

### PR DESCRIPTION
**Pull request by M374LX**
Workaround for #5809.

### CODE REVIEW

File: src/core/audio/AudioSdl.cpp [hunk 1]
```diff
@@ -99,6 +99,9 @@ AudioSdl::AudioSdl( bool & _success_ful, Mixer*  _mixer ) :
 	_success_ful = true;
 
 #ifdef LMMS_HAVE_SDL2
+	// Workaround for a race condition that causes SDL to segfault
+	SDL_Delay(250);
```
Comment (PhysSong) [inline src/core/audio/AudioSdl.cpp:103]: Is 250ms a really good time interval? I'm not sure, but you can try lowering this and try to reproduce the crash.

File: src/core/audio/AudioSdl.cpp [hunk 2]
```diff
@@ -99,6 +99,9 @@ AudioSdl::AudioSdl( bool & _success_ful, Mixer*  _mixer ) :
 	_success_ful = true;
 
 #ifdef LMMS_HAVE_SDL2
+	// Workaround for a race condition that causes SDL to segfault
```
Comment (JohannesLorenz) [inline src/core/audio/AudioSdl.cpp:102]: Using a delay to fix a race condition sounds generally strange. But if it's the only option...

Can you give any source where you found the solution? Or is this just fixed out of your mind?

Comment (M374LX) [reply src/core/audio/AudioSdl.cpp:102]: The idea of adding a delay came out of my mind.

Audio devices are implemented as threads in SDL. Consequently, inserting a delay in theory gives the the playback device's thread time to "finish starting" before the capture device (and thread) is created, which prevents a race condition.

Comment (JohannesLorenz) [reply src/core/audio/AudioSdl.cpp:102]: Well, if it improves things, why not...

**Comment by JohannesLorenz** [comment]
I'm going to run some tests now, and then we can get this merged.

**Comment by JohannesLorenz** [comment]
Thanks for the PR! The patch seems to cause no regression, I merged it.

**Pull request merged**